I1 i12 datasetdocumentation localchanges

solveigvandervegt opened the pull request:
I have documented the data sets contact_home, contact_work, contact_school, contact_other, cases, mort_sever_default, and population. The latter three are defaults that, for as far as I can see are at the moment always overwritten by data that comes from the excel files. I couldn't identify any code where age_categories and countries_demog are used. I think these are pulled form the excel file as these excel files contain country, specific data.

solveigvandervegt committed:
Create data.R

Documentation for the datasets in data_RComo

solveigvandervegt committed:
Added final data sets

This file now contains documentation for contact_home, contact_work, contact_school, contact_other, cases, population, and mort_server_default. The last three are loaded as defaults, but are mostly overwritten by data from excel files.

solveigvandervegt committed:
Deleted excessive files

Documentation for these data sets have been consolidated into the data.R file

ben18785 commented on R/data.R:
```diff
@@ -0,0 +1,104 @@
+#' contact_home
+#' 
+#' A collection of contact matrices for 152 countries, describing which age 
```
I would be explicit: "A list of 16x16 contact matrices..."

ben18785 commented on R/data.R:
```diff
@@ -0,0 +1,104 @@
+#' A collection of contact matrices for 152 countries, describing which age 
+#' groups have contact with which other age groups AT HOME. The units of entries
+#' are the average number of contacts with a person in that age group in a day.
+#' The columns are the age groups 0-5 (X1), 5-10 (X2), 10-15 (X3), etc. up to 80 years old. 
```
It might be good to be explicit: "10-15 (X3), and continuing with 5 year gaps up until 80 years old."

ben18785 commented on R/data.R:
```diff
@@ -0,0 +1,104 @@
+
+#' contact_school
+#' 
+#' A collection of contact matrices for 152 countries, describing which age 
```
Same as above re: list and 16x16 matrices

oscardm20994 replied:
Some of the processing of these contact matrices that Sumali and I are looking at involve a transformation from 16x16 to 21x21 to match the age ranges from the excel file. I think it's the 21x21 versions that finally get parsed to the ode solver so might be worth noting this change in dimensions somewhere. It may not be most appropriate here though and if so no worries!

oscardm20994 replied:
this transformation occurs in the covidage_v16.8.R script so might not need to mention it here

ben18785 commented on R/data.R:
```diff
@@ -0,0 +1,104 @@
+#' A collection of contact matrices for 152 countries, describing which age 
+#' groups have contact with which other age groups AT SCHOOL. The units of entries
+#' are the average number of contacts with a person in that age group in a day.
+#' The columns are the age groups 0-5 (X1), 5-10 (X2), 10-15 (X3), etc. up to 80 years old. 
```
Same as above

ben18785 commented on R/data.R:
```diff
@@ -0,0 +1,104 @@
+
+#' contact_work
+#' 
+#' A collection of contact matrices for 152 countries, describing which age 
```
Same as above

oscardm20994 replied:
Might be worth mentioning that these matrices are not symmetrical (as you'd expect them to be) as they were taken from survey data of different age groups. Not a big issue but if a user loads the data and does a quick plot of it they might be surprised by its asymmetry.

ben18785 commented on R/data.R:
```diff
@@ -0,0 +1,104 @@
+#' A collection of contact matrices for 152 countries, describing which age 
+#' groups have contact with which other age groups AT WORK. The units of entries
+#' are the average number of contacts with a person in that age group in a day.
+#' The columns are the age groups 0-5 (X1), 5-10 (X2), 10-15 (X3), etc. up to 80 years old. 
```
Same

ben18785 commented on R/data.R:
```diff
@@ -0,0 +1,104 @@
+
+#' contact_other
+#'  
+#' A collection of contact matrices for 152 countries, describing which age 
```
Same (sorry, I know this is repetitive!).

ben18785 commented on R/data.R:
```diff
@@ -0,0 +1,104 @@
+#' groups have contact with which other age groups AT OTHER PLACES (not covered
+#' by the other contact matrices). The units of entries
+#' are the average number of contacts with a person in that age group in a day.
+#' The columns are the age groups 0-5 (X1), 5-10 (X2), 10-15 (X3), etc. up to 80 years old. 
```
Same

ben18785 commented on R/data.R:
```diff
@@ -0,0 +1,104 @@
+
+#' population
+#' 
+#' Per country and per age group the number of individuals in the population, 
```
Per country and per age group: the number of individuals in the population; the birth rate (what units?); and the death rate (what units?).

ben18785 commented on R/data.R:
```diff
@@ -0,0 +1,104 @@
+#' }
+"mort_sever_default"
+
+#' population
```
Do we know where this dataset comes from? If not, I think we should ask Bo, perhaps.

ben18785 commented on R/data.R:
```diff
@@ -0,0 +1,104 @@
+
+#' cases
+#' 
+#' NOTE: I think this data is actually imported from the excel file. The sheets
```
If it is typically imported by the Excel, shall we remove it? (We can always add it in later if needed.)

ben18785 reviewed (changes requested):
Thanks @solveigvandervegt -- looking really good! I've added a few inline comments that'd be great if you could have a look at. I also think that maybe we should remove "cases", "age_categories" and "countries_demog" since these aren't used (if it turns out they are required later, we can add them back in). 

Re: tests, I think you may need to build the documentation locally via `devtools::document()` since I don't think the changes in your documentation has made its way through to the "man" directory. Also, did you check for style using the `lintr::lint_package()` function? I tried running this locally on this branch and it highlighted a few style suggestions. We have just added code styling checks (see the "Contributing" section of the README).

ben18785 commented:
@rcw5890 @alisterde Any idea what's happening with the lintr here? It looks like it may be failing because of the `pillar` dependency of `dplyr`.

rcw5890 commented:
Hi @solveigvandervegt and @ben18785 
The URL which failed in the lintr check is working fine, which makes me suspect some temporary outage on the r-project website. I'm going to make a commit removing some trailing spaces to trigger the check again.

rccreswell committed:
remove some spaces

rcw5890 commented:
Lintr is working again now. The problems it found appear as annotations in the code [here](https://github.com/Como-DTC-Collaboration/comoDTC/pull/33/files). Apparently, the transgressions aren't considered serious enough for the check to fail.

ben18785 commented:
@rcw5890 Thanks for looking at that. Yep, must have been a temporary unavailability of the package, or something.

oscardm20994 commented on R/data.R:
```diff
@@ -55,11 +55,50 @@
+#' out by default. However, the excel file only provides data for one
+#' country, not for all like this data set does.
+#' 
+#' Data on the number of cases and deaths per country per day from 31-12-2019.
```
is it data for just this date (for initialising purposes) ? or this day onwards (for model fitting purposes)?

If it's this day onwards when does the data run until?

oscardm20994 reviewed (commented):
Hi @solveigvandervegt this looks in really good shape and is definitely a really useful tool to a user. Have added a few minor comments of specific lines about the contact matrices and the epidemiology data. I'd definitely agree with Ben regarding the removal of data that appears to be overwritten by the excel file. It should make it significantly easier to track the data that's going into the model.

solveigvandervegt commented:
@ben18785 @oscardm20994 Thank you both for your comments :) I will work on incorporating those, hopefully before the end of the week!

solveigvandervegt committed:
Delete superfluous data files and update data description

Deleted the data files population, cases, country_demog, age_categories, and mort_sever_default as these are not used within the code. The necessary data is instead imported from the excel files.

Update the description of the contact matrices to improve clarity and fit style guidelines.

ben18785 committed:
Built docs

addresses #33

ben18785 committed:
Merge branch 'master' into i1-i12-datasetdocumentation-localchanges

ben18785 reviewed (approved):
Thanks @solveigvandervegt -- looks really great! I'm happy for this to go in now.

oscardm20994 commented:
Sorry for the delay, These changes all look good and am also more than happy for this to merge (which you may have already done). Great stuff @solveigvandervegt
